zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1010,9 +1011,11 @@ private void validateCreateRequest(String path, CreateMode createMode, Request r
             if (id == null || id.getScheme() == null) {
                 throw new KeeperException.InvalidACLException(path);
             }
-            if (id.getScheme().equals("world") && id.getId().equals("anyone")) {
+            if (id.getScheme().equals("world") && id.getId().equals("anyone") && !QuorumPeerConfig
+                .isSetX509ClientIdAsAclEnabled()) {
```

Comment:
nit, isSetX509ClientIdAsAclEnabled -> isX509ClientIdAsAclEnabled ?

Reply:
I think "set X509 client id as acl" makes sense and less ambiguous. For example, Kafka has a similar client-side config called `zookeeper.set.acl`. See [this](https://docs.confluent.io/2.0.0/kafka/zookeeper-authentication.html). Let me know if there's a strong reason this needs to be changed?

Reply:
Changed to setX509ClientIdAsAclEnabled